compiler/rustc_attr_parsing/src/parser.rs

```diff
@@ -127,7 +127,7 @@ impl<'a> ArgParser<'a> {
             }
             AttrArgs::Eq { eq_span, expr } => Self::NameValue(NameValueParser {
                 eq_span: *eq_span,
-                value: expr_to_lit(dcx, &expr),
+                value: expr_to_lit(dcx, &expr, value.span().unwrap()),
```

Comment:
You're very possibly introducing a new ICE here. Could you just pass this as an `Option<Span>` and `.unwrap_or(DUMMY_SP)` when it's needed?